File: terry/programmers/brute-force/exam.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> f = {1,2,3,4,5};
vector<int> s = {2,1,2,3,2,4,2,5};
vector<int> t = {3,3,1,1,2,2,4,4,5,5};

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    
    vector<int> score(3,0);
    
    for(int i = 0; i < answers.size(); i++) {
        int a = answers[i];
        if(f[i%f.size()] == a) score[0]++;
        if(s[i%s.size()] == a) score[1]++;
        if(t[i%t.size()] == a) score[2]++;
    }
    
    int max = 0;
    for(int i = 0; i < 3; i++) {
        max = max > score[i] ? max : score[i];
    }
    
    for(int i = 0; i < 3; i++) {
        if(score[i] == max) answer.push_back(i+1);
    }
    
    return answer;
}

File: taehwa/programmers/brute-force/exam.ts
export default function solution(answers: number[]) {
  const patterns = [
    [1, 2, 3, 4, 5],
    [2, 1, 2, 3, 2, 4, 2, 5],
    [3, 3, 1, 1, 2, 2, 4, 4, 5, 5],
  ];

  const scores = patterns.map((pattern) => {
    let score = 0;

    answers.forEach((answer, index) => {
      if (answer === pattern[index % pattern.length]) score++;
    });

    return score;
  });

  const highestScore = Math.max(...scores);

  const result: number[] = [];

  scores.forEach((score, index) => {
    if (score === highestScore) {
      result.push(index + 1);
    }
  });

  return result.sort((a, b) => a - b);
}
